server/src/main/java/org/opensearch/querygroup/LabelingHeader.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.querygroup;
+
+/**
+ * Main enum define various headers introduced by {@link org.opensearch.plugins.LabelingPlugin}s
+ */
+public enum LabelingHeader {
+    QUERY_GROUP_ID("queryGroupId");
```

Comment:
Ideally this enum should be part of core as it is used by the `RequestLabelingService`, but it has plugin implementation specific values. Hence, I am slightly confused whether this should be in core or part of plugin

Reply:
I think this should be part of core since these headers will be consumed in core.